### PR TITLE
🎉 New Source: IBM DB2

### LINKED ISSUE #2975: New Source: IBM DB2
## Tell us about the new integration you’d like to have
https://www.ibm.com/analytics/db2

## Describe the context around this new integration
* Which team in your company wants this integration, what for? This helps us understand the use case.
* How often do you want to run syncs?
* If this is an API source connector, which entities/endpoints do you need supported?

## Describe the alternative you are considering or using
What are you considering doing if you don’t have this integration through Airbyte?

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1200367912513076/1200368226088171) by [Unito](https://www.unito.io)

Todo:
1. Generate (use a template generator) jdbc connector and implement connections 
(https://www.ibm.com/docs/en/db2-for-zos/11?topic=samples-example-simple-jdbc-application, 
https://www.ibm.com/docs/en/db2-for-zos/11?topic=hjacds-connecting-data-source-using-drivermanager-interface-data-server-driver-jdbc-sqlj) 
2. Create unit test 
3. Create Integration test (container for tests https://www.testcontainers.org/modules/databases/db2/) 
4. Create Comprehensive test:
    - Investigate and prepare use cases (https://www.ibm.com/support/producthub/db2/docs/content/SSEPGG_11.5.0/com.ibm.db2.luw.sql.ref.doc/doc/r0008483.html?pos=2): 
    - Implement comprehensive tests.
    - Prepare pull request, pass all checks.

### PR DESCRIPTION
**Pull request by irynakruk**
## What
New source to integrate with IBM Db2

## How
1. Generated jdbc connector and implemented connections based on Db2 jdbc driver
2. Created unit tests
3. Created Integration tests (container for tests https://www.testcontainers.org/modules/databases/db2/)
4. Created Comprehensive tests

## Recommended reading order
1. `x.java`
2. else

## Pre-merge Checklist

<details><summary> <strong> Connector checklist </strong> </summary>
<p>

- [x] Issue acceptance criteria met
- [x] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Secrets are annotated with `airbyte_secret` in output spec
- [x] Unit & integration tests added as appropriate (and are passing)
    * Community members: please provide proof of this succeeding locally e.g: screenshot or copy-paste acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [x] `/test connector=connectors/<name>` command as documented [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-an-existing-connector) is passing. 
    * Community members can skip this, Airbyters will run this for you. 
- [x] Code reviews completed
- [ ] Credentials added to Github CI if needed and not already present. [instructions for injecting secrets into CI](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#using-credentials-in-ci). 
- [x] Documentation updated 
    - [x] README
    - [x] CHANGELOG.md
    - [x] Reference docs in the `docs/integrations/` directory.
    - [x] Build status added to [build page](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/builds.md)
- [x] Build is successful
- [ ] Connector version bumped like described [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-a-connector)
- [x] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-a-connector)
- [ ] No major blockers
- [ ] PR merged into master branch
- [ ] Follow up tickets have been created
- [ ] Associated tickets have been closed & stakeholders notified
</p>
</details>

### CODE REVIEW

**Review by yaroslav-hrytsaienko — APPROVED** [review]
@irynakruk  great pr! Meanwhile i did not found the table with use cases for comprehensive data types. Can you please add it with a comment to a ticket? Thanks!

**Comment by irynakruk** [comment]
> @irynakruk great pr! Meanwhile i did not found the table with use cases for comprehensive data types. Can you please add it with a comment to a ticket? Thanks!

@yaroslav-hrytsaienko Thanks for fast review!
Sure, I added table with use cases directly to the ticket:
https://github.com/airbytehq/airbyte/issues/2975#issuecomment-863061884

File: airbyte-integrations/connectors/source-db2/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Changelog
+
+## 0.1.0
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-db2/CHANGELOG.md:3]: this now lives in the docs page, see [this example](https://docs.airbyte.io/integrations/sources/stripe#changelog)

**Review by sherifnada — COMMENTED** [review]
we should also add this to `docs/integrations/README.md` as a `beta` connector

**Review by sherifnada — APPROVED** [review]
LGTM, nicely done @irynakruk !

File: airbyte-integrations/connectors/source-db2/CHANGELOG.md (hunk 1 above)
Comment (irynakruk) [reply airbyte-integrations/connectors/source-db2/CHANGELOG.md:3]: Thanks, I've updated

**Comment by mrmelon-xyz** [comment]
Hi there, Is there any plan to add CDC support to this connector?

**Comment by sherifnada** [comment]
@mrmelon-xyz there is no timeline at the moment

**Pull request merged**